.github/workflows/pull-request.yml

```diff
@@ -0,0 +1,51 @@
+name: Pull request comment
+
+on: pull_request
+
+permissions:
+  pull-requests: write
+  contents: write
+
+jobs:
+  Explore-GitHub-Actions:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - name: Get changed files
+        id: changed-files
+        uses: tj-actions/changed-files@v14.6
+        with:
+          files: icons/*.svg
+      - name: Generate comment
+        id: generate-comment
+        run: |
+          delimiter="$(openssl rand -hex 8)"
+          echo "body<<$delimiter" >> $GITHUB_OUTPUT
+          for file in ${{ steps.changed-files.outputs.all_changed_files }}; do
+            cat "$file" | # get file content
+            tr '\n' ' ' | # remove line breaks
+            sed -e 's/<svg[^>]*>/<svg>/g' | # remove attributes from svg element
+            base64 -w 0 | # encode svg
+            sed "s|.*|<img width=\"50%\" alt=\"$file\" src=\"https://lucide.dev/api/gh-icon/&.svg\"/>|"
```

Comment:
Not totally sure about creating an API endpoint for this purpose, couldn't we just as easily delegate the generation to a script and embed the base64 encoded SVG files into the `src` attribute?

```html
<img src="data:image/svg+xml;base64,${encodedSvgContent}"/>
```

That way we also would only have to add the dependencies as dev deps.

Reply:
Sadly, GitHub does not support `data`-URLs.

Reply:
Oh, damn, it does indeed not. :(
And evidently Github's API [doesn't allow uploading images either](https://github.com/peter-evans/create-or-update-comment/issues/68). :(

Reply:
Yes and it's a bit of a security issue if we would run this inside public forks.
Maybe in the future we can have a create/upload icon button on the website that shows you a preview and possible errors.

Reply:
A possible alternative approach I found through some Googling: we can add a dedicated repository branch and upload the image files there via the [repository contents API](https://docs.github.com/en/rest/repos/contents?apiVersion=2022-11-28#create-or-update-file-contents).

My greatest issue with this approach is that we can't really cache dynamic content on the site, as it became evident in #896.

@ericfennis: what do you think?

Reply:
There shouldn't be any caching problems since this doesn't use the file system in any way, right?

Reply:
Well, the issue is exactly that it _should_ be cached, so that the site doesn't have to generate it over and over again.

Adding an appropriate cache control header preferably with an obscenely large `max-age` should in theory work: https://vercel.com/guides/how-to-configure-the-cache-control-response-header-in-vercel-projects#using-edge-functions

Reply:
Cache control will work if the content for each request changes, it will bypass the settled cache. So I think it's ok to use our API for it.